v2/pkg/operators/common/dsl/dsl.go

```diff
@@ -915,6 +917,72 @@ func init() {
 
 			return buf.String(), nil
 		}),
+		"resolve": makeDslWithOptionalArgsFunction("(host string, format string) string", func(args ...interface{}) (interface{}, error) {
+			format := "4"
+			dnsType := dns.TypeA
+			if len(args) > 1 {
+				format = strings.ToLower(types.ToString(args[1]))
+			}
+
+			switch format {
+			case "4", "a":
+				dnsType = dns.TypeA
+			case "6", "aaaa":
+				dnsType = dns.TypeAAAA
+			case "cname":
+				dnsType = dns.TypeCNAME
+			case "ns":
+				dnsType = dns.TypeNS
+			case "txt":
+				dnsType = dns.TypeTXT
+			case "srv":
+				dnsType = dns.TypeSRV
+			case "ptr":
+				dnsType = dns.TypePTR
+			case "mx":
+				dnsType = dns.TypeMX
+			case "soa":
+				dnsType = dns.TypeSOA
+			case "caa":
+				dnsType = dns.TypeCAA
+			}
+
+			dnsx.DefaultOptions.QuestionTypes = []uint16{dnsType}
+			dnsClient, err := dnsx.New(dnsx.DefaultOptions)
```

Comment:
The implementation is functional. Anyway, I suggest using retryabledns directly to avoid bringing in all dnsx dependencies. Additionally, it will allow you to specify the DNS query type directly. The code would be similar to this:
```go
dnsClient, err := dnsclientpool.Get(nil, nil)
...
rawResp, err := dnsClient.Query(types.ToString(args[0]), dnsType)
...
```